### PR TITLE
Cut v2.0.0

### PR DESCRIPTION
**Pull request by lilic**
🎉 

cc @mrueg @brancz @tariq1890 

/hold

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -153,24 +151,25 @@ http_request_duration_seconds_count{handler="metrics",method="get"} 30
...
-> Note: These recommendations are based on scalability tests done over a year ago. They may differ significantly today.
-
 Resource usage for kube-state-metrics changes with the Kubernetes objects (Pods/Nodes/Deployments/Secrets etc.) size of the cluster.
 To some extent, the Kubernetes objects in a cluster are in direct proportion to the node number of the cluster.
 
-As a general rule, you should allocate
+As a general rule, you should allocate:
 
-* 200MiB memory
+* 250MiB memory
```
Comment (lilic) [inline README.md:159]: I based these off OpenShift tests and the resources we set in the scale tests.

File: README.md [hunk 2]
```diff
@@ -70,7 +68,7 @@ At most, 5 kube-state-metrics and 5 [kubernetes releases](https://github.com/kub
 |--------------------|---------------------|---------------------|----------------------|----------------------|-----------------------|
 | **v1.8.0**         |         -           |          -           |          -           |          -           |          -           |
 | **v1.9.8**         |         ✓           |          -           |          -           |          -           |          -           |
-| **v2.0.0-rc.1**    |         -           |          -/✓         |         -/✓          |          ✓           |          ✓           |
```
[the remark this answers is not in the document]
Comment (lilic) [reply README.md:73]: We haven't tested or bumped k8s deps, I would prefer to do this in a follow up as soon as 2.0.0 is released so not to postpone this release any longer, wdyt?

File: README.md [hunk 3]
```diff
@@ -153,24 +151,25 @@ http_request_duration_seconds_count{handler="metrics",method="get"} 30
...
 
-As a general rule, you should allocate
+As a general rule, you should allocate:
 
-* 200MiB memory
+* 250MiB memory
 * 0.1 cores
 
-For clusters of more than 100 nodes, allocate at least
+Note that if CPU limits are set too low, kube-state-metrics' internal queues will not be able to be worked off quickly enough, resulting in increased memory consumption as the queue length grows. If you experience problems resulting from high memory allocation, try increasing the CPU limits.
```
[the remark this answers is not in the document]
Comment (lilic) [reply README.md:162]: Yeah. But this was there before I just reordered it :) do you propose to remove this or reword it?

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mrueg) [reply README.md:73]: Yes, that is fine.
 I just assume folks will ask "is 1.21 is supported?" (we could also add -/- )

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mrueg) [reply README.md:162]: Sorry I didn't spot that it changed the location. Maybe something along the lines of:

```suggestion
Note that if CPU limits are set too low, kube-state-metrics' internal queues will not be able to be worked off quickly enough, resulting in increased memory consumption as the queue length grows. If you experience problems resulting from high memory allocation or CPU throttling, try increasing the CPU limits.
```

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (lilic) [reply README.md:73]: I suspect more questions will then be "why is it not supported" :D

[the remark this answers is not in the document]
Comment (mrueg) [reply README.md:73]: Fair point. :D Okay, let's leave it as is :+1:

**Comment by lilic** [comment]
@mrueg please have another look, tests passed 🎉

File: README.md [hunk 4]
```diff
@@ -153,24 +151,25 @@ http_request_duration_seconds_count{handler="metrics",method="get"} 30
...
-* 200MiB memory
+* 250MiB memory
 * 0.1 cores
 
-For clusters of more than 100 nodes, allocate at least
+Note that if CPU limits are set too low, kube-state-metrics' internal queues will not be able to be worked off quickly enough, resulting in increased memory consumption as the queue length grows. If you experience problems resulting from high memory allocation or CPU throttling, try increasing the CPU limits.
 
-* 2MiB memory per node
-* 0.001 cores per node
+### Latency
```
Comment (mrueg) [inline README.md:164]: Do you want to add this to the ToC as well?

Comment (lilic) [reply README.md:165]: Done thanks!

**Comment by tariq1890** [comment]
🚀 🚀

**Review by tariq1890 — APPROVED** [review]

**Comment by lilic** [comment]
/hold cancel

https://kubernetes.io/blog/2021/04/13/kube-state-metrics-v-2-0/ 🎉

**Pull request merged**